src/briefcase/bootstraps/base.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import Any, TypedDict
+
+
+class AppContext(TypedDict):
+    formal_name: str
+    app_name: str
+    class_name: str
+    module_name: str
+    project_name: str
+    description: str
+    author: str
+    author_email: str
+    bundle: str
+    url: str
+    license: str
+    briefcase_version: str
+    template_source: str
+    template_branch: str
+
+
+class BaseGuiBootstrap(ABC):
+    """Definition for a plugin that defines a new Briefcase app."""
+
+    fields: list[str] = [
+        "app_source",
+        "app_start_source",
+        "pyproject_table_briefcase_additional",
+        "pyproject_table_briefcase_app_additional",
+        "pyproject_requires",
+        "pyproject_test_requires",
+        "pyproject_table_macOS",
+        "pyproject_table_linux",
+        "pyproject_table_linux_system_debian",
+        "pyproject_table_linux_system_rhel",
+        "pyproject_table_linux_system_suse",
+        "pyproject_table_linux_system_arch",
+        "pyproject_table_linux_appimage",
+        "pyproject_table_linux_flatpak",
+        "pyproject_table_windows",
+        "pyproject_table_iOS",
+        "pyproject_table_android",
+        "pyproject_table_web",
+        "pyproject_extra_content",
+    ]
```

Comment:
I'm not a huge fan of having to define the context fields here....since this needs to stay in sync with the methods below as well as the list in `cookiecutter.json`....but they need to be enumerated somewhere so they can be arbitrarily iterated by Briefcase.

Furthermore, though, this allows a third-party plugin to override the supported fields and implement entirely different methods that'd Briefcase run to build `context`.